easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -328,6 +328,11 @@ def configure_step(self, srcdir=None, builddir=None):
         # ensure CMake uses EB python, not system or virtualenv python
         options.update(get_cmake_python_config_dict())
 
+        # pass the preferred host compiler to the CUDA compiler
+        cuda_root = get_software_root('CUDA')
+        if cuda_root:
+            self.cfg.update('preconfigopts', 'CUDAHOSTCXX=%s' % which(os.getenv('CXX', 'g++')))
```

Comment:
My understanding is `CUDACXX` should be `nvcc` in this case, and we would have to map our CUDA variable to `CUDAARCHS` so that it looks like `"35;50;72"` (see https://cmake.org/cmake/help/latest/prop_tgt/CUDA_ARCHITECTURES.html#prop_tgt:CUDA_ARCHITECTURES)

Reply:
There is [`CUDAFLAGS`](https://cmake.org/cmake/help/latest/envvar/CUDAFLAGS.html) too, but I don't think we have any value for that, it will probably be package specific.

Reply:
Sure, I can look into setting those as well.

Reply:
Done in https://github.com/easybuilders/easybuild-easyblocks/pull/3523/commits/0857b9fbcab2f975408aa59623d128aa75e00d5c. I suppose that the architecture variable may be a little bit more dangerous than the other ones, as projects themselves could also define/override this (?).